reactive/vault_kv.py

```diff
@@ -55,9 +55,11 @@ def manage_app_kv_flags():
 def update_app_kv_hashes():
     try:
         app_kv = vault_kv.VaultAppKV()
-        if hookenv.is_leader() and app_kv.any_changed():
+        if hookenv.is_leader():
             # force hooks to run on non-leader units
             hookenv.leader_set({'vault-kv-nonce': host.pwgen(8)})
+        if app_kv.any_changed():
+            # Update the local unit hashes at successful exit
```

Comment:
These are the actual changes to address LP1949807

Reply:
See PR https://github.com/juju-solutions/layer-vault-kv/pull/8 in which I made it so only leaders call update_hashes (oops) to fix a hook error.

I need to test to make sure that the original hook error bug isn't being reintroduced by this... stay tuned.

Reply:
Ahh yes, that would more than likely occur when trying to run
```python
    def update_hashes(self):
        """
        Update the hashes in Vault, thus marking all fields as unchanged.

        This is done automatically at exit.
        """
        log("Writing hashes to {}", self._hash_path)
       self._client.write(self._hash_path, **self._new_hashes)
```

where the value of `self._new_hashes = {}`.  I believe this would cause the `hvac` client to produce this error.  You're essentially writing no updates to a specific kv path. 

My change shouldn't have this issue because of the use of 
```python 
        if app_kv.any_changed():
            # Update the local unit hashes at successful exit
            app_kv.update_hashes()
```

`any_changed` **sort-of** confirms `_new_hashes` has something in it.  The only case i'm not sure about is if the hash value is ever `None`  -- which i'm not sure how that would ever be looking at the code.

Reply:
Yep, that seems right. I didn't encounter any issues with this in a test deployment so I think we should be good.

Reply:
@Cynerva , i'm having second (maybe third) thoughts about the setting of

```python
        if hookenv.is_leader():
            # force hooks to run on non-leader units
            hookenv.leader_set({'vault-kv-nonce': host.pwgen(8)})
```
on every hook at exit on the leader.  Doesn't this say 
> if the hook is exiting, and we're the leader unit, run `leader_set vault-kv-none = <new password>`

If so, maybe it should keep it's wrapping

```python
if hookenv.is_leader() and app_kv.any_changed():
```